### PR TITLE
docs, snap: enumerate snap types correctly in docs and comments

### PR DESCRIPTION
**Pull request by stevenwilkin**

### CODE REVIEW

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

File: docs/rest.md [hunk 1]
```diff
@@ -253,7 +253,7 @@ Sample result:
     * `version`: a string representing the version.
     * `revision`: a number representing the revision.
     * `icon`: a url to the snap icon, possibly relative to this server.
-    * `type`: the type of snap; one of `app`, `framework`, `kernel`,
+    * `type`: the type of snap; one of `app`, `core`, `kernel`,
```
Comment (mvo5) [inline docs/rest.md]: we also have `gadget`

Comment (mvo5) [inline docs/rest.md]: Ups, silly me

**Comment by mvo5** [comment]
:+1:

**Comment by stevenwilkin** [comment]
Rebased

**Pull request merged**